backend/src/akvo/lumen/component/keycloak.clj

```diff
@@ -120,6 +191,64 @@
                        (assoc request-draft
                               :body (json/encode {"emailVerified" true})))))
 
+(defn do-promote-user-to-admin
+  [{:keys [api-root] :as keycloak} tenant author-claims user-id]
+  (if (= (get author-claims "sub") user-id)
+    (http/bad-request {"reason" "Tried to alter own tenant role"})
+    (let [request-draft (request-draft keycloak)
+          tenant-group-id (get (group-by-path
+                                keycloak request-draft tenant) "id")
+          admin-group-id (get (group-by-path
+                               keycloak request-draft (format "%s/admin" tenant))
+                              "id")]
+      (if (and (= 204 (add-user-to-group request-draft api-root user-id
+                                         admin-group-id))
+               (= 204 (remove-user-from-group request-draft api-root user-id
+                                              tenant-group-id)))
+        (http/ok (fetch-user-by-id request-draft api-root tenant user-id))
+        (do
+          (println (format "Tried to promote user: %s" user-id))
+          (http/internal-server-error))))))
```

Comment:
Same question for the two subsequent places where we have the same branching logic: Is the `else` part reachable?